Device/Device.cs

```diff
@@ -66,7 +69,43 @@ public async Task DisconnectDevice()
             await mqttClient.DisconnectAsync();
         }
 
-        public MqttApplicationMessage ConstructMessage(string topic, string payload, bool retainFlag = false)
+        public async Task ConnectDeviceUsingCACertificate()
+        {
+            var certificates = new List<X509Certificate>() { new X509Certificate2("CA-Certificate.pfx", "1234") };
+            var options = new MqttClientOptionsBuilder()
+                .WithTcpServer(hubAddress, 8883)
+                .WithCredentials(new MqttClientCredentials() { Username = hubAddress + "/" + deviceId })
+                .WithClientId(deviceId)
+                .WithProtocolVersion(MqttProtocolVersion.V311)
+                .WithTls(new MqttClientOptionsBuilderTlsParameters
+                {
+                    UseTls = true,
+                    Certificates = certificates
+                })
+                .Build();
+
+            await mqttClient.ConnectAsync(options, CancellationToken.None);
+        }
+
+        public async Task ConnectDeviceUsingSelfSignedCertificate()
+        {
+            var certificates = new List<X509Certificate>() { new X509Certificate2("SelfSigned-Certificate.pfx", "1234") };
+            var options = new MqttClientOptionsBuilder()

```

Comment:
The options above and here are very similar, perhaps, we should make a method that accepts certificate as a parameter?